mkt/webapps/tasks.py

```diff
@@ -383,45 +382,85 @@ def export_data(name=None):
     created_files = []
-    for f in files:
-        template = loader.get_template('webapps/dump/apps/' + f)
-        dest = os.path.join(settings.DUMPED_APPS_PATH, f)
-        open(dest, 'w').write(template.render(context))
-        created_files.append(f)
+    for filename in files:
+        template = loader.get_template('webapps/dump/apps/%s' % filename)
+        dest = os.path.join(target_directory, filename)
+        local_storage.open(dest, 'w').write(template.render(context))
+        created_files.append(filename)
     return created_files
 
 
 @task
-def compress_export(filename, files):
-    # Note: not using storage because all these operations should be local.
-    target_dir = os.path.join(settings.DUMPED_APPS_PATH, 'tarballs')
-    target_file = os.path.join(target_dir, filename + '.tgz')
+def compress_export(tarball_name, date):
+    # We need a temporary directory on the local filesystem that will contain
+    # all files in order to call `tar`.
+    local_source_dir = tempfile.mkdtemp()
+
+    apps_dirpath = os.path.join(settings.DUMPED_APPS_PATH, 'apps')
+
+    # In case there are no 'apps' directory. It should not happen in prod, but
+    # it's nice to have it to prevent the task from failing entirely.
+    with private_storage.open(
+            os.path.join(apps_dirpath, '0', '.keep'), 'w') as fd:
+        fd.write('.')
+
+    # Now, copy content from private_storage to that temp directory. We don't
+    # need to worry about creating the directories locally, the storage class
+    # does that for us.
+    for dirpath, dirnames, filenames in walk_storage(apps_dirpath):
+        for filename in filenames:
+            src_path = os.path.join(dirpath, filename)
+            dst_path = os.path.join(
+                local_source_dir, 'apps', os.path.basename(dirpath), filename)
+            copy_to_storage(
+                src_path, dst_path, src_storage=private_storage,
+                dst_storage=local_storage)
+
+    # Also add extra files to the temp directory.
+    extra_filenames = compile_extra_files(local_source_dir, date)
+
+    # All our files are now present locally, let's generate a local filename
+    # that will contain the final '.tar.gz' before it's copied over to
+    # public storage.
+    local_target_file = tempfile.NamedTemporaryFile(
+        suffix='.tgz', prefix='dumped-apps-')
+
+    # tar ALL the things!
+    cmd = ['tar', 'czf', local_target_file.name, '-C',
+           local_source_dir] + ['apps'] + extra_filenames
+    task_log.info(u'Creating dump {0}'.format(local_target_file.name))
+    subprocess.call(cmd)
 
-    if not os.path.exists(target_dir):
-        os.makedirs(target_dir)
+    # Now copy the local tgz to the public storage.
+    remote_target_filename = os.path.join(
+        settings.DUMPED_APPS_PATH, 'tarballs', '%s.tgz' % tarball_name)
+    copy_to_storage(local_target_file.name, remote_target_filename,
+                    dst_storage=public_storage)
```

Comment:
`src_storage` defaults to `local_storage` so like above I didn't set arguments I didn't need to set